### PR TITLE
added more tests and refactored

### PR DESCRIPTION
**Pull request by josephmancuso**
Might fail tests until we get Masonite released

### CODE REVIEW

**Comment by josephmancuso** [comment]
The code coverage decreased so much because you were previously covering the tests directory when it should have been the app directory.

**Comment by vaibhavmule** [comment]
Thanks for the PR and fixing the coverage!

File: app/http/controllers/ArticleController.py [hunk 1]
```diff
@@ -52,14 +53,14 @@ def index(self):
         return {'articles': list_of_articles, 'articlesCount': len(list_of_articles)}
 
     def feed(self):
-        users = self.request.user().followed_users().lists('user_id').serialize()
-        articles = Article.with_('author', 'favorites').where_in('author_id', users)
+        users = self.user.followed_users().lists('user_id').serialize()
+        articles = Article.with_('author', 'favorites').where_in('author_id', users).get()
```
Comment (vaibhavmule) [inline app/http/controllers/ArticleController.py]: can you do `order_by` after `.get()`

```suggestion
        articles = Article.with_('author', 'favorites').where_in('author_id', users)
```

File: app/http/controllers/AuthController.py [hunk 2]
```diff
@@ -7,8 +7,11 @@
...
-        user_data = request.input('user')
-        if auth.once().login(user_data['email'], user_data['password']):
-            user = User.where('email', user_data['email']).first()
+        email = request.input('user.email')
+        password = request.input('user.password')
+        if auth.once().login(email, password):
+            user = User.where('email', email).first()
             user.generate_token()
             return {'user': user.serialize()}
+
```
Comment (vaibhavmule) [inline app/http/controllers/AuthController.py]:
```suggestion
        request.status(400)
```

**Review by vaibhavmule — CHANGES_REQUESTED** [review]

File: app/http/controllers/ArticleController.py (hunk 1 above)
Comment (vaibhavmule) [reply app/http/controllers/ArticleController.py]: I am adding tests for it lets see what happens.

**Review by vaibhavmule — APPROVED** [review]

**Pull request merged**